index.js

```diff
@@ -1,8 +1,25 @@
-// TODO: remove web3 requirement
-// Call functions directly on the provider.
-var Web3 = require("web3");
-
 var Blockchain = {
+
+  getBlockByNumber: function(blockNumber, provider, callback){
+    var params = [blockNumber, true];
+    provider.sendAsync({
+      jsonrpc: '2.0',
+      method: 'eth_getBlockByNumber',
```

Comment:
So the JSON RPC spec suggests that `blockNumber` needs to be an integer - have you tested this to be otherwise?

Reply:
Good question! Initially tried with `0` and it didn't work. Also looks like `web3` is converting the blockNumber to hex [here](https://github.com/ethereum/web3.js/blob/develop/lib/web3/formatters.js#L53-L60),  where they implement the same endpoint.